src/query/storages/system/src/columns_table.rs

```diff
@@ -138,7 +143,20 @@ impl ColumnsTable {
                 .list_tables(tenant.as_str(), database.name())
                 .await?
             {
-                for field in table.schema().fields() {
+                let mut fields = table.schema().fields().clone();
+                if table.engine() == VIEW_ENGINE {
+                    if let Some(query) = table.options().get(QUERY) {
+                        let mut planner = Planner::new(ctx.clone());
+                        let (plan, _, _) = planner.plan_sql(query).await?;
+                        let schema = infer_table_schema(&plan.schema()).unwrap();
+                        fields = schema.fields().clone();
+                    } else {
+                        return Err(ErrorCode::Internal(
+                            "Logical error, View Table must have a SelectQuery inside.",
+                        ));
+                    }
+                }
```

Comment:
```suggestion
                let fields = if table.engine() == VIEW_ENGINE {
                    if let Some(query) = table.options().get(QUERY) {
                        let mut planner = Planner::new(ctx.clone());
                        let (plan, _, _) = planner.plan_sql(query).await?;
                        let schema = infer_table_schema(&plan.schema())?;
                        schema.fields().clone()
                    } else {
                        return Err(ErrorCode::Internal(
                            "Logical error, View Table must have a SelectQuery inside.",
                        ));
                    }
                } else {
                    table.schema().fields().clone()
                };
```

Reply:
Thank you so much for the code snippet! Updated.